### DISCUSSION TITLE
How to specify different schema and/or table name for Hasura metadata

### DISCUSSION THREAD

**Discussion by botwhytho** [hasura/graphql-engine#9668] [Question] [answered]
The managed DB team at work seems very unlikely to support us using Hasura if we need the meta-data table/tables stored in a different schema that is not `public` as unfortunately even the `admin` user is rather locked down and inflexible. How can I define which schema to store the hasura metadata info in? It may not live in the same database as our data, but it unfortunately MUST be in a `public` schema so hoping this is something I've yet to find in docs and not a feature request.

**Comment by adas98012** [comment]
@botwhytho 
You can have a separate metadata (postgres) database (distinct from your data). 
Here is the [doc](https://hasura.io/docs/latest/deployment/graphql-engine-flags/config-examples/#add-metadata-database)

**Comment by botwhytho** [comment]
Hi, I've seen these docs before, whether specifying the same or a
different database server,  there seems to be no settings for specifying a
different schema than 'hdb_catalog' where the metadata tables will live.

Can someone confirm if this is an impossibility at the moment? Per
description above I may not have the flexibility to create that schema or
tables in any other schema other than 'public'. Unfortunate but actual
situation at work and can't imagine I've been the only person to bump into
this

**Comment by adas98012** [comment, marked as answer]
@botwhytho 
As per our docs

`The Hasura GraphQL Engine when initialized, creates a schema called hdb_catalog in the Postgres database and initializes a few tables under it. This schema and the internal tables are generally termed as the metadata catalogue and is responsible to manage the internal state of the Hasura GraphQL Engine.`

So, it won't be possible to use a custom schema name for metadata catalog.
